homeassistant/components/webostv/media_player.py

```diff
@@ -206,6 +207,9 @@ def update_sources(self):
                 or any(word in app["id"] for word in conf_sources)
             ):
                 self._source_list["Live TV"] = app
+        if not self._source_list and source_list:
+            self._source_list = source_list
+        source_list = {}
```

Comment:
The method ends here, you do not need to reset source_list

Reply:
you are right, I was struggling with that a little bit, should I free the list implicitly or not